hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/OmSnapshotLocalDataManager.java

```diff
@@ -195,6 +228,7 @@ void addVersionNodeWithDependents(OmSnapshotLocalData snapshotLocalData) throws
   }
 
   private void init() throws IOException {
+    this.locks = omMetadataManager.getHierarchicalLockManager();
```

Comment:
I'd like to have an explanation in ReadOnlyHierarchicalResourceLockManager for when it is used and when PoolBasedHierarchicalResourceLockManager is used, because I don't think I'll remember in the future.